Codelab test failure: SemanticException: Cannot compare 'java.sql.Date' to 'java.lang.String'

From main branch Cloud Build tests:
https://pantheon.corp.google.com/cloud-build/builds/998b9517-2452-4a27-99ab-63a9b142e795;step=1?project=cloud-spanner-hibernate-ci
```
Step #1: [INFO] -------------------------------------------------------
Step #1: [INFO]  T E S T S
Step #1: [INFO] -------------------------------------------------------
Step #1: [INFO] Running codelab.AppIT
Step #1: [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 32.35 s <<< FAILURE! -- in codelab.AppIT
Step #1: [ERROR] codelab.AppIT.test -- Time elapsed: 32.25 s <<< ERROR!
Step #1: java.lang.IllegalArgumentException: org.hibernate.query.SemanticException: Cannot compare left expression of type 'java.sql.Date' with right expression of type 'java.lang.String'
Step #1: 	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:143)
Step #1: 	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:167)
Step #1: 	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:173)
Step #1: 	at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:860)
Step #1: 	at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:765)
Step #1: 	at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:140)
Step #1: 	at codelab.App.readData(App.java:85)
Step #1: 	at codelab.App.main(App.java:52)
Step #1: 	at codelab.AppIT.test(AppIT.java:34)
Step #1: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Step #1: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
Step #1: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Step #1: 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
Step #1: 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
Step #1: 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
Step #1: 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
Step #1: 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
Step #1: 	at org.junit.contrib.java.lang.system.internal.LogPrintStream$1$1.evaluate(LogPrintStream.java:30)
Step #1: 	at org.junit.contrib.java.lang.system.internal.PrintStreamHandler$3.evaluate(PrintStreamHandler.java:48)
Step #1: 	at org.junit.contrib.java.lang.system.internal.LogPrintStream$1.evaluate(LogPrintStream.java:26)
Step #1: 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
Step #1: 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
Step #1: 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
Step #1: 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
Step #1: 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
Step #1: 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
Step #1: 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
Step #1: 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
Step #1: 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
Step #1: 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
Step #1: 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
Step #1: 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
Step #1: 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:316)
Step #1: 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:240)
Step #1: 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:214)
Step #1: 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:155)
Step #1: 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
Step #1: 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
Step #1: 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
Step #1: 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
Step #1: Caused by: org.hibernate.query.SemanticException: Cannot compare left expression of type 'java.sql.Date' with right expression of type 'java.lang.String'
Step #1: 	at org.hibernate.query.sqm.internal.TypecheckUtil.assertComparable(TypecheckUtil.java:358)
Step #1: 	at org.hibernate.query.sqm.tree.predicate.SqmComparisonPredicate.<init>(SqmComparisonPredicate.java:48)
Step #1: 	at org.hibernate.query.sqm.tree.predicate.SqmComparisonPredicate.<init>(SqmComparisonPredicate.java:34)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.createComparisonPredicate(SemanticQueryBuilder.java:2562)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitComparisonPredicate(SemanticQueryBuilder.java:2504)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitComparisonPredicate(SemanticQueryBuilder.java:275)
Step #1: 	at org.hibernate.grammars.hql.HqlParser$ComparisonPredicateContext.accept(HqlParser.java:6432)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitWhereClause(SemanticQueryBuilder.java:2356)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitWhereClause(SemanticQueryBuilder.java:275)
Step #1: 	at org.hibernate.grammars.hql.HqlParser$WhereClauseContext.accept(HqlParser.java:6173)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitQuery(SemanticQueryBuilder.java:1254)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitQuerySpecExpression(SemanticQueryBuilder.java:1035)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitQuerySpecExpression(SemanticQueryBuilder.java:275)
Step #1: 	at org.hibernate.grammars.hql.HqlParser$QuerySpecExpressionContext.accept(HqlParser.java:2132)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitSimpleQueryGroup(SemanticQueryBuilder.java:1020)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitSimpleQueryGroup(SemanticQueryBuilder.java:275)
Step #1: 	at org.hibernate.grammars.hql.HqlParser$SimpleQueryGroupContext.accept(HqlParser.java:2003)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitSelectStatement(SemanticQueryBuilder.java:490)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.visitStatement(SemanticQueryBuilder.java:449)
Step #1: 	at org.hibernate.query.hql.internal.SemanticQueryBuilder.buildSemanticModel(SemanticQueryBuilder.java:322)
Step #1: 	at org.hibernate.query.hql.internal.StandardHqlTranslator.translate(StandardHqlTranslator.java:71)
Step #1: 	at org.hibernate.query.internal.QueryInterpretationCacheStandardImpl.createHqlInterpretation(QueryInterpretationCacheStandardImpl.java:145)
Step #1: 	at org.hibernate.query.internal.QueryInterpretationCacheStandardImpl.resolveHqlInterpretation(QueryInterpretationCacheStandardImpl.java:132)
Step #1: 	at org.hibernate.internal.AbstractSharedSessionContract.interpretHql(AbstractSharedSessionContract.java:802)
Step #1: 	at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:852)
Step #1: 	... 36 more
Step #1: 
Step #1: [INFO] 
Step #1: [INFO] Results:
Step #1: [INFO] 
Step #1: [ERROR] Errors: 
Step #1: [ERROR]   AppIT.test:34 Â» IllegalArgument org.hibernate.query.SemanticException: Cannot compare left expression of type 'java.sql.Date' with right expression of type 'java.lang.String'
```

--- a/google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java
+++ b/google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java
@@ -82,7 +82,10 @@ public class App {
 
   private static void readData(Session session) {
     List<Singer> singers =
-        session.createQuery("from Singer where birthDate >= '1990-01-01' order by lastName").list();
+        session
+            .createQuery(
+                "from Singer where birthDate >= cast('1990-01-01' as date) order by lastName")
+            .list();
     System.out.println("Singers who were born in 1990 or later:");
     for (Singer singer : singers) {
       System.out.println(

